compiler/build.gradle

```diff
@@ -54,47 +54,68 @@ def addLibraryIfNotLinked = { libName, argList ->
 }
 
 model {
-  toolChains {
-    gcc(Gcc) {
-      target("x86_64") {
-        cppCompiler.withArguments { args ->
-          args << "-m64"
-        }
-        linker.withArguments { args ->
-          args << "-m64"
-        }
+  if (System.getProperty('use.vc') != null) {
+    toolChains {
+      visualCpp(VisualCpp) {
       }
-      target("x86_32") {
-        cppCompiler.withArguments { args ->
-          args << "-m32"
-        }
-        linker.withArguments { args ->
-          args << "-m32"
-        }
+    }
+    platforms {
+      // VisualCpp supports 'x86' and 'amd64', but it builds 'x86' by default.
+      // Thus Protobuf is likely to have been built for x86. No matter what the
+      // detected architecture is, 'x86' is the safest target.
+      x86_32 {
+        architecture 'x86'
+      }
+      x86_64 {
+        architecture 'x86'
+      }
+      local_arch {
+        architecture 'x86'
       }
-      target('local_arch') { }
     }
-    clang(Clang) {
-      target("x86_64") {
-        cppCompiler.withArguments { args ->
-          args << "-m64"
+  } else {
+    toolChains {
+      gcc(Gcc) {
+        target("x86_64") {
+          cppCompiler.withArguments { args ->
+            args << "-m64"
+          }
+          linker.withArguments { args ->
+            args << "-m64"
+          }
         }
-        linker.withArguments { args ->
-          args << "-m64"
+        target("x86_32") {
+          cppCompiler.withArguments { args ->
+            args << "-m32"
+          }
+          linker.withArguments { args ->
+            args << "-m32"
+          }
         }
+        target('local_arch') { }
+      }
+      clang(Clang) {
+        target("x86_64") {
+          cppCompiler.withArguments { args ->
+            args << "-m64"
+          }
+          linker.withArguments { args ->
+            args << "-m64"
+          }
+        }
+        target('local_arch') { }
       }
-      target('local_arch') { }
-    }
-  }
-  platforms {
-    x86_32 {
-      architecture "x86_32"
-    }
-    x86_64 {
-      architecture "x86_64"
     }
-    local_arch {
-      architecture 'local_arch'
+    platforms {
+      x86_32 {
+        architecture "x86_32"
```

Comment:
It works because we have defined it in the toolChains block. For platform strings, I intend to use the architecture strings emitted by osdetector, instead of the ones defined by a particular build system, because we want to use a unified glossary across the artifact producing and consuming processes, and across all build systems (typically Maven and Gradle)